draft-ietf-quic-transport.md

```diff
@@ -2611,173 +2612,74 @@ CONNECTION_CLOSE or RST_STREAM frame. Error codes share a common code space.
 
-QUIC_ERROR_MIGRATING_PORT (0x80000056):
-: There was an error encountered migrating port only.
+: An endpoint uses this with CONNECTION_CLOSE to signal that the connection is
+  being closed abruptly in the absence of any error.  An endpoint uses this with
+  RST_STREAM to signal that the stream is no longer wanted or in response to the
+  receipt of a RST_STREAM for that stream.
 
-QUIC_EMPTY_STREAM_FRAME_NO_FIN (0x80000032):
-: We received a STREAM_FRAME with no data and no fin flag set.
+CANCELLED (0x80000004):
 
-QUIC_FLOW_CONTROL_RECEIVED_TOO_MUCH_DATA (0x8000003b):
-: The peer received too much data, violating flow control.
+: An endpoint sends this with RST_STREAM to indicate that the stream is not
+  wanted and that no application action was taken for the stream.  This error
+  code is not valid for use with CONNECTION_CLOSE.
 
-QUIC_FLOW_CONTROL_SENT_TOO_MUCH_DATA (0x8000003f):
-: The peer sent too much data, violating flow control.
+FLOW_CONTROL_ERROR (0x80000005):
 
-QUIC_FLOW_CONTROL_INVALID_WINDOW (0x80000040):
-: The peer received an invalid flow control window.
+: An endpoint received more data than it permitted in its advertised data limits
+  (see {{flow-control}}.
 
-QUIC_CONNECTION_IP_POOLED (0x8000003e):
-: The connection has been IP pooled into an existing connection.
+STREAM_ID_ERROR (0x80000006):
 
-QUIC_TOO_MANY_OUTSTANDING_SENT_PACKETS (0x80000044):
-: The connection has too many outstanding sent packets.
+: An endpoint received a frame for a stream identifier that exceeded its
+  advertised maximum stream ID.
 
-QUIC_TOO_MANY_OUTSTANDING_RECEIVED_PACKETS (0x80000045):
-: The connection has too many outstanding received packets.
+STREAM_STATE_ERROR (0x80000007):
 
-QUIC_CONNECTION_CANCELLED (0x80000046):
-: The QUIC connection has been cancelled.
+: An endpoint received a frame for a stream that was not in a state that
+  permitted that frame (see {{stream-states}}).
 
-QUIC_BAD_PACKET_LOSS_RATE (0x80000047):
-: Disabled QUIC because of high packet loss rate.
+FINAL_OFFSET_ERROR (0x80000008):
 
-QUIC_PUBLIC_RESETS_POST_HANDSHAKE (0x80000049):
-: Disabled QUIC because of too many PUBLIC_RESETs post handshake.
+: An endpoint received a STREAM frame containing data that exceeded the
+  previously established final offset.  Or an endpoint received a RST_STREAM
+  frame containing a final offset that was lower than the maximum offset of data
+  that was already received.  Or an endpoint received a RST_STREAM frame
+  containing a different final offset to the one already established.
 
-QUIC_TIMEOUTS_WITH_OPEN_STREAMS (0x8000004a):
-: Disabled QUIC because of too many timeouts with streams open.
+FRAME_FORMAT_ERROR (0x80000009):
 
-QUIC_TOO_MANY_RTOS (0x80000055):
-: QUIC timed out after too many RTOs.
+: An endpoint received a frame that was badly formatted.  For instance, an empty
+  STREAM frame that omitted the FIN flag, or an ACK frame that has more
+  acknowledgment ranges than the remainder of the packet could carry.
```

Comment:
These are very different errors in terms of what a peer would do with this information. I don't think collapsing these two examples is a good idea, since the peer can actually do something concrete with this information -- separating STREAM_FRAME_FORMAT_ERROR from ACK_FRAME_FORMAT_ERROR is a useful for monitoring and debugging.